pkg/reconciler/gc/v1/gc.go

```diff
@@ -30,28 +30,36 @@ import (
 	"knative.dev/serving/pkg/apis/serving"
 	v1 "knative.dev/serving/pkg/apis/serving/v1"
 	clientset "knative.dev/serving/pkg/client/clientset/versioned"
-	configreconciler "knative.dev/serving/pkg/client/injection/reconciler/serving/v1/configuration"
 	listers "knative.dev/serving/pkg/client/listers/serving/v1"
 	configns "knative.dev/serving/pkg/reconciler/gc/config"
 )
 
-// reconciler implements controller.Reconciler for Garbage Collection resources.
-type reconciler struct {
-	client clientset.Interface
-
-	// listers index properties about resources
+// GC stores the types needed for garbage collection
+type GC struct {
+	client         clientset.Interface
 	revisionLister listers.RevisionLister
+	config         *v1.Configuration
 }
 
-// Check that our reconciler implements configreconciler.Interface
-var _ configreconciler.Interface = (*reconciler)(nil)
+// MakeGC is a factory function to make a GC struct
+func MakeGC(
```

Comment:
```suggestion
func NewGC(
```

Reply:
But more importnat, do we even need the state? Just pass the params to the Collect?